Build doesn't work when compiling with GCC on MacOS (Clang).

Following the instructions on the README - I encountered this error on the second step in building.

ninja -C build
ninja: Entering directory `build'
[2/4] Compiling C object 'src/0...a18c720@@0x01@exe/control.c.o'
FAILED: src/0x01/a18c720@@0x01@exe/control.c.o 
cc -Isrc/0x01/a18c720@@0x01@exe -Isrc/0x01 -I../src/0x01 -Xclang -fcolor-diagnostics -pipe -Werror -fno-stack-protector -Wno-implicit-function-declaration -MD -MQ 'src/0x01/a18c720@@0x01@exe/control.c.o' -MF 'src/0x01/a18c720@@0x01@exe/control.c.o.d' -o 'src/0x01/a18c720@@0x01@exe/control.c.o' -c ../src/0x01/control.c
../src/0x01/control.c:33:7: error: incompatible pointer types passing 'char (*)[64]' to parameter of type 'char *' [-Werror,-Wincompatible-pointer-types]
        gets(&buffer);
             ^~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/stdio.h:168:18: note: passing argument to parameter here
char    *gets(char *);
                    ^
1 error generated.
[3/4] Linking target src/0x00/0x00.
ninja: build stopped: subcommand failed.

--- a/src/0x02/control.c
+++ b/src/0x02/control.c
@@ -30,7 +30,7 @@ int main(int argc, char **argv)
 	puts("Overflow the buffer! Set the return address.");
 	printf("The return address for the shell function is @ %p\n", &shell);
 
-	gets(&buffer);
+	gets(buffer);
 
 	return 0;
 }

